### ISSUE TITLE
Printing loss after every batch

### ISSUE THREAD

**Issue by parulML** [pytorch/ignite#552]
I want to print loss after completion of every batch and I am using below code for the same but it's not working the way I am expecting. Can anyone please suggest me that what am I doing wrong.

Thanks in advance.

``` 
def print_logs(engine, dataloader, mode, history_dict):
    train_evaluator.run(dataloader)
    metrics = train_evaluator.state.metrics
    avg_loss = metrics['loss']
    avg_dice = metrics['dice']
    avg_iou = metrics['iou']
    print(mode + " Results - Epoch {} - Avg loss: {:.2f} Avg dice: {:.2f} Avg iou: {:.2f}"
          .format(engine.state.iteration, avg_loss, avg_dice, avg_iou))
    for key in train_evaluator.state.metrics.keys():
        history_dict[key].append(train_evaluator.state.metrics[key])

trainer.add_event_handler(Events.ITERATION_COMPLETED, print_logs, dataloader, 'Training', training_history)
trainer.add_event_handler(Events.ITERATION_COMPLETED, print_logs, dataloader_val, 'Validation', validation_history) ```

**Comment by vfdev-5** [comment]
@parulML if you would like to print training batch loss during the training phase this can be done as 
```python
# Some background preparations :
def batch_processing_function(engine, batch):
    # ...
    # Model weights update
    #
    return (batch_loss.item(), another_batch_metric.item(), ...)

trainer = Engine(batch_processing_function)

@trainer.on(Events.ITERATION_COMPLETED)
def print_batch_output(engine):
    output = engine.state.output
    # output is what is returned by batch_processing_function
    # output = (batch_loss.item(), another_batch_metric.item(), ...)
    print("Batch Loss: {} \t another batch metric: {}".format(output[0], output[1]))

```
More on the conepts can be found [here](https://pytorch.org/ignite/concepts.html)
If you would like to log batch output with tqdm or tensorboard logger, please take a look at 
- https://pytorch.org/ignite/contrib/handlers.html#module-ignite.contrib.handlers.tqdm_logger
- https://pytorch.org/ignite/contrib/handlers.html#module-ignite.contrib.handlers.tensorboard_logger

HTH

**Comment by vfdev-5** [comment]
@parulML I close the issue as answered. Feel free to reopen if you still have a trouble with this answer. Thanks
